lib/AST/Decl.cpp

```diff
@@ -9464,6 +9464,25 @@ GenericTypeParamDecl *OpaqueTypeDecl::getExplicitGenericParam(
   return genericParamType->getDecl();
 }
 
+bool OpaqueTypeDecl::exportUnderylingTypeToClients() const {
+  auto mod = getDeclContext()->getParentModule();
+  if (mod->getResilienceStrategy() != ResilienceStrategy::Resilient)
+    return true;
+
+  ValueDecl *namingDecl = getNamingDecl();
+  if (auto *AFD = dyn_cast<AbstractFunctionDecl>(namingDecl)) {
+    return AFD->getResilienceExpansion() == ResilienceExpansion::Minimal;
+
+  } else if (auto *ASD = dyn_cast<AbstractStorageDecl>(namingDecl)) {
+    return !ASD->isResilient();
+
+  } else {
+    llvm_unreachable("The naming decl is expected to be either an AFD or ASD");
+  }
+
+  return false;
```

Comment:
Minor, just personal preference, but I would find this block more readable and in line with project style if it leaned in to  the early returns being exists, and therefore eliding `else`s as noise:

```suggestion
  if (auto *AFD = dyn_cast<AbstractFunctionDecl>(namingDecl))
    return AFD->getResilienceExpansion() == ResilienceExpansion::Minimal;

  if (auto *ASD = dyn_cast<AbstractStorageDecl>(namingDecl))
    return !ASD->isResilient();

  llvm_unreachable("The naming decl is expected to be either an AFD or ASD");
```

Reply:
Good idea. I applied it to the new version.